### ISSUE TITLE
Unable to split back flowSet into flowFrames

### ISSUE THREAD

**Issue by KonradZych** [HelenaLC/CATALYST#57]
Hi, I tried analysing multiple FCS files at once and then splitting the resulting flowSet into flowFrames for further analysis. I am not able to do that unfortunately. The problems seems to be in some kind of weird formatting of the FileNum column.

1. I run the code to read and concat the FCS files (I tried to read and analyse files one by one but this didn't work either as `normCytof` fails if I supply flowFrame instead of flowSet):
``` 
filesToRead <- list.files(pattern="*.FCS")
ff <- concatFCS(filesToRead, file_num = TRUE)
```

2. I normalise:
```
normFCS <- normCytof(x=ff, y="dvs", k=500, plot=TRUE,remove_beads=TRUE)
```

3. I then tried to split the flowSet back into flowFrames but couldn't figure it out, so I decided to have a look at the fileNum column and just didi a simple plot:
```
concat_beads@exprs[,60] # column 60 is the last one and contains the FileNum
```

The result is a bit tricky:
![image](https://user-images.githubusercontent.com/606363/64563010-93acdc80-d34e-11e9-8051-90122e2eaa02.png)

So as you can see the FileNum does contain some kind of a continuous variable instead of the discreet number for each file. I cannot wrap my head around this fully but it seems to me that this column got transformed in some way?

**Comment by SamGG** [comment]
Hi. Just a simple thought. Maybe you should store the filenum column as a vector instead of concatenating itin the flowframe data.

**Comment by KonradZych** [comment]
Thanks, @SamGG but this could not work, since after beads-normalization, the beads are removed from the data and therefore the vector would not be aligned anymore with flowSet. The other problem I noticed is that the mapping of file names to the fileNum is not clear (I cannot retrieved information which file number relates to which file name)

**Comment by HelenaLC** [comment]
This is indeed a bug; thanks for bringing it to my attention with such a clear example, making it really easy to follow & fix. I have just pushed a fix to CATLYST 1.9.9 on Bioc devel (3.10), which will become the new release and of the month. `normCytof` now assures thatt the `fileNum` column is left untransformed. 
(In case this is unclear, you can install the package from the GitHub master branch, or through Bioc via `BiocManager::install("CATALYS", version = "devel")`)

```
# using package example data, CATALYST v1.9.9
library(CATALYST)
data(raw_data)
ff <- concatFCS(raw_data, file_num = TRUE)
ff <- normCytof(x = ff, y = "dvs", k = 120, plot = FALSE)
plot(ff[[1]]@exprs[, "FileNum"])
```
![image](https://user-images.githubusercontent.com/14542264/66568640-287f4180-eb6a-11e9-8e2c-261f326c7968.png)
